### PR TITLE
Fix duplicate entry errors on IDN_CLAIM tables when two APIM servers are started in parallel

### PR DESCRIPTION
**Pull request by msm1992**
### Proposed changes in this pull request
This PR adds the fix for https://github.com/wso2/product-apim/issues/7914.

### CODE REVIEW

File: components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/dao/ClaimDialectDAO.java [hunk 1]
```diff
@@ -136,4 +159,42 @@ public void removeClaimDialect(ClaimDialect claimDialect, int tenantId) throws C
...
+     * In mssql, constraint violation error is wrapped in an SQLServerException instead of an
+     * SQLIntegrityConstraintViolationException. So for mssql we are checking the error code of the
+     * exception thrown to identify constrant violation errors in mssql.
+     *
+     * @param e SQL exception caught
+     * @return true if the exeption caught is an SQL Integrity Constraint violation
+     */
+    private boolean isSQLIntegrityConstraintViolation(SQLException e) {
+
+        return e instanceof SQLIntegrityConstraintViolationException
```
Comment (IsuraD) [inline components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/dao/ClaimDialectDAO.java:196]: This should be supported for all the supported db types and versions.

File: components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/exception/DuplicateClaimException.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        super(errorDescription);
+    }
+
+    public DuplicateClaimException(String errorDescription, Throwable cause) {
+
+        super(errorDescription, cause);
+    }
+
+}
```
Comment (IsuraD) [inline components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/exception/DuplicateClaimException.java:36]: need to add a new line

**Review by IsuraD — CHANGES_REQUESTED** [review]
Please check the comments

**Comment by DMHP** [comment]
@msm1992 
Thanks for the contribution. I will close the PR until the comments resolved. Please re-open once it is resolved.

**Pull request closed without merging**